include/sys/types.h

```diff
@@ -133,8 +133,8 @@ typedef _size_t      rsize_t;
  * gid_t is used for group IDs.
  */
 
-typedef int16_t      uid_t;
-typedef int16_t      gid_t;
+typedef unsigned int uid_t;
```

Comment:
Should this be conditional based on `CONFIG_SMALL_MEMORY`?

Reply:
Done